schema.sql

```diff
@@ -0,0 +1,56 @@
+CREATE TABLE users (
+	-- users contains all user-data. An entry in this table corresponds to one
+	-- specific person. The account balance is reconstructed completely out of the
+	-- transactions table, to reduce duplication of information.
+
+
+	-- user_id is a sequential identifier.
+	user_id INTEGER NOT NULL,
+	-- name is the username used for display and login.
+	name TEXT,
+	-- password is a bcrypt-hashed password.
+	password BINARY,
+
+	-- constraints
+	PRIMARY KEY (user_id)
+);
+
+CREATE TABLE cards (
+	-- cards contains all card-data. An entry in this table corresponds to one
+	-- physical card. Every user can have an arbitrary number of cards.
+
+
+	-- card_id is a sequential identifier.
+	card_id BINARY NOT NULL,
```

Comment:
ja, aber gefühlt fände ich es schöner, INTEGER statt BINARY oder TEXT als pirmary key zu nehmen. und zumindest in sqlite (die einzige datenbank, die ich bisher bedient habe, und ich weiß nicht, ob man statt 'ner einfachen datei irgendwie 'nen ganzen server aufsetzen will…) ist primary key integer auch das einzige was typesafe ist, wenn ich mich nicht irre